### DISCUSSION TITLE
Full chemistry L58 run

### DISCUSSION THREAD

**Discussion by JulioTBacmeister** [NCAR/amwg_dev#74] [TF1: Unified physics for L58/L93]
We want to do a full chemistry L58 run output ozone fields (and more??) for future L58 development runs.  

What configuration should we use? Choices:

1) "control" without additional diffusion
2) Version with additional diffusion above convection (upto 50hPa)

Option 2 has somewhat better UTLS water vapor in tropics, but has a larger cold bias than option 1.  Zonal mean T from both runs is shown below. I would suggest we start with option 1, since the physics mod in option 2 is not quite "ready-for-prime-time".

Zonal mean T from control (issue #53)
![image](https://user-images.githubusercontent.com/9221710/153651144-6d6699d6-82ae-4942-9c30-9b1b9108c7b3.png)

Zonal mean T from run w/ added diffusion (issue #60)
![image](https://user-images.githubusercontent.com/9221710/153651416-0792c771-04d7-4c7b-8266-f2bb9d70550a.png)

**Comment by tilmes** [comment]
Probably better to not have a large cold bias for running with chemistry, that could change the ozone fields a bit and with that the radiation, Simone

**Reply by dan800** [reply]
Part of the bias in the tropical stratosphere could be that the QBO phases aren't matching (though 10 years you'd hope they averaged out). It is worrying that a cold pole bias has appeared. Thought we'd fixed that, but maybe that was just WACCM. @JulioTBacmeister Is this a result of not dumping GW momentum at the lid?

**Reply by JulioTBacmeister** [reply]
@dan800 I don't remember ever seeing a bias-free zonal mean T from this model at high lats. I suppose it could be related to the radiation condition for param GW at the top.

And as far as the tropical bias I think it is likely due to the hard wired 50hPa top of the ad-hoc diffusion applied above convection.

**Reply by JulioTBacmeister** [reply]
In any case, I suggest we go with the "control" option 1

**Reply by tilmes** [reply]
@cecilehannay Could you confirm that the version we got from you is the one to use (option 1)?

**Reply by cecilehannay** [reply]
@tilmes: Edited on Feb 16 to fix an error in the run directory. 
The correct run directory for pg3 is:
 /glade/p/cesmdata/cseg/runs/cesm2_0/f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001

----

Option 1 is the run: 
https://github.com/NCAR/amwg_dev/issues/53

The case directory is at:
```
 /glade/p/cesmdata/cseg/runs/cesm2_0/f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001
```

It uses the tag cam6_3_041 available at: 
```
/glade/work/hannay/cesm_tags/cam6_3_041/
```

The CAM_CONFIG_OPTS included:
```
./xmlchange CAM_CONFIG_OPTS="-phys cam_dev -age_of_air_trcs -chem waccm_sc_mam4 -nlev 58"
```

The user_nl_cam included:
```
ncdata = '/glade/p/cesm/amwg_dev/juliob/FWsc_ne30pg3_58L_GRID_48_taperstart10km_lowtop_BL10_v3_beta1p75_Top_43km.nc'
se_rsplit            = 3
se_nsplit           = 2
se_hypervis_subcycle =  4
```

**Reply by dan800** [reply]
I think you would want to delete the -age_of_air_trcs line, since this run will be less than a decade long.

**Reply by cecilehannay** [reply]
FYI: I gave you what was used in: ```f.e21.FWscHIST.ne30pg2_L48_BL10_cam6_3_041_control.hf.001```, but as Dan said we don't include ```-age_of_air_trcs``` in more recent runs.

**Reply by adamrher** [reply]
Also just a reminder to explicitly set ```-microphys mg2``` in the CONFIG_OPTS, since that will omit two additional tracers that don't matter at our workhorse resolutions. You want to place it after the ```-phys cam_dev``` argument to be safe, since cam_dev invokes the unnecessary mg3 tracers.

**Reply by tilmes** [reply]
@cecilehannay This run you pointed me to is a hybrid run from 1989 but starting in 1979, is that intended?

**Comment by cecilehannay** [comment]
Yes. This is the way we have been setting our runs. We do an hybrid run starting in 1979 from the end of another run (i.e. 1989) to get a more spin up land. We have to add a flag in the user_nl_clm so it ignores the year. 

But for the atmosphere we explicitly set the initial condition with ncdata.

**Reply by tilmes** [reply]
where do I find the restart files?

**Comment by cecilehannay** [comment]
The restart files are in inputdata. They should come out of the box.

**Reply by tilmes** [reply]
This directory is empty: /glade/p/cesmdata/cseg/inputdata/ccsm4_init/f.e21.FWscHIST_BGC.ne30_ne30_mg17_L48_revert-J.001

**Reply by adamrher** [reply]
I don't mean to muddy the waters, but @tilmes I setup a run-script to recreate Cecile's configuration here: 
```
/glade/work/aherring/CESM2/arh-git-scripts/run-scripts/FWscHIST.sh
```

Obviously the fincl's are out of date. I've actually never got it to work for a branch run, and so it's currently set-up for a startup run (but see commented out restart variables as something to start with). It would be nice to get the exact same restarts that Cecile's using ... If the branch is supposed to run out of the box, then ```GET_REFCASE=T``` and we need the name of the refcase. Otherwise ```GET_REFCASE=F``` and we need to stage the restarts and rpointer in the run directory.

**Reply by dan800** [reply]
Adding more mud - don't you have to worry about land model spin up? I think we also need a spun up finidat/ncdata for that component as well.

**Reply by tilmes** [reply]
Land should be spun up, I am producing my own case, need fincls for the oxidant fields, and NOx deposition as well. And I add some chemistry  so we can look at other tracers. If you have specific output you want me to add beyond Cecile's case, please let me know. The ncdata file will be from a WACCM FCASE (CMIP6).

**Reply by cecilehannay** [reply]
The files are in: 
/glade/p/cesmdata/cseg/inputdata/cesm2_init/f.e21.FWscHIST_BGC.ne30_ne30_mg17_L48_revert-J.001/1989-01-01

**Reply by cecilehannay** [reply]
@dan800: we wanted to use the same atm initial condition for all the runs. Julio interpolated a single file to every vertical grid so everything is consistent. 

The land was coming from a L48 run. We used that for every run.

**Reply by dan800** [reply]
OK, as long as the shrubs are fully grown ;-)

**Reply by cecilehannay** [reply]
@dan800: fully grown shrubs going all the way to the stratosphere. ;)

**Reply by tilmes** [reply]
@adamrher I think you helped Jun to interpolate a chemistry file to the 58 levels? I tried to follow the steps, added P0 to my IC file and then applied these commands: ncremap --vrt=/glade/p/cesm/amwg_dev/juliob/FWsc_ne30pg3_58L_GRID_48_taperstart10km_lowtop_BL10_v3_beta1p75_Top_43km.nc f.e21.FWHISTBgcCrop.f09_f09_mg17.CMIP6-AMIP-WACCM.001.cam.i.1979-01-01-00000.nc f.e21.FWHISTBgcCrop.f09_f09_mg17.CMIP6-AMIP-WACCM.001.cam.i.1979-01-01.58L.nc
But I am getting this error: 
/glade/u/apps/dav/opt/nco/4.9.5/gnu/9.1.0/bin/ncremap: line 2801: 193891 Segmentation fault      ncks -O -t 2 --no_tmp_fl --hdr_pad=10000 --gaa remap_script=ncremap --gaa remap_hostname=crhtc58 --gaa remap_version=4.9.5 --vrt="/glade/p/cesm/amwg_dev/juliob/FWsc_ne30pg3_58L_GRID_48_taperstart10km_lowtop_BL10_v3_beta1p75_Top_43km.nc" "/glade/p/cesm/chwg_dev/tilmes/inputdata/init/cam7/1979-01-01-00000/f.e21.FWHISTBgcCrop.f09_f09_mg17.CMIP6-AMIP-WACCM.001.cam.i.1979-01-01-00000.nc" "f.e21.FWHISTBgcCrop.f09_f09_mg17.CMIP6-AMIP-WACCM.001.cam.i.1979-01-01.58L.nc"
ncremap: ERROR Failed to vertically interpolate. cmd_rgr[0] failed.

Do you  have a suggestion how to fix this?

**Reply by adamrher** [reply]
obviously not a very informative error message. But thinking out loud, the cmd should be ```ncremap --vrt=<template> in.nc out.nc```. And I think the issue is that the template you're using has the correct target vertical grid ... But your ```in.nc``` is on a different horizonal grid (f09) then the template ... seems like this could be an issue. You moght want to try to map the ```in.nc``` file to the ne30pg3 grid first, or try to find a CAM-CHEM ```in.nc``` that is already on the ne30pg3 grid.

**Reply by tilmes** [reply]
Oh yes, thanks, I guess, I forgot to first interpolate to the ne30pg3 grid!

**Reply by adamrher** [reply]
I would suggests setting ```--alg_typ=conserve``` or ```--alg_typ=bilinear``` if you are using the default nco/4.9.5 to do horizontal remapping; Charlie switched the default remapping to use an "nco remapping" in going from nco/4.7.9 -> nco/4.9.5, which Gary and I found is problematic and can corrupt the output. So either use nco/4.7.9 for horizontal remaps, where ESMF remapping is the default, or else invoke ```--alg_typ=...``` to force nco/4.9.5 to use ESMF remapping.

**Comment by tilmes** [comment]
The chemistry run is ready to start (I ran a year without a problem and will briefly check it before running the next 20 years): /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/CAM7_tests/f.e21.FWscHIST_chemistry.ne30pg2_L48_BL10_cam6_3_041_control.hf.001
I had to remove some fincls that were in Cecile's run, because the model  did not find them, not sure if that is a problem? DQCORE, CMFMCDZ, DQCORE, V200, V850. I am also checking with Mike to be sure to have the forcing output correct. The model will run 4  years  a day, so the 20 years should be done in 5 days after I start.

**Reply by dan800** [reply]
@tilmes Has the issue of specifying the upper boundary condition on this run from a prior CAM-CHEM or WACCM (pref.) run be resolved? If not, I think this run will be compromised and it would be good to do the best job we can on this new vertical grid.

@lkemmons @JulioTBacmeister @cecilehannay @fvitt @drmikemills

**Reply by tilmes** [reply]
No, it has  not been resolved or included at this point.

**Reply by dan800** [reply]
IMO, I would wait to run the 20 years until it has. I would like to see the UBC in this low top run for NOx, CFCs, CH4 and H2O be taken from a WACCM run.

**Reply by tilmes** [reply]
@dan800 CAMchem usually does not include UBC as far as I am aware of.. we could also do a second run later with the UBC, and compare.. depending on how fast we can resolve this

**Reply by islasimpson** [reply]
It would also be useful to know how to do this for the needs of this Scripps seasonal prediction project.  As of now, I have run 7 years with L83 with CAM6 and I've left the upper boundary, which I believe is no flux for T, 1e-6 for H2O and 2e-8 for CH4, although at this point I'm not advecting methane so I'm not sure if that's doing anything.  I haven't had a chance to look at it yet, but I will do today.

**Reply by adamrher** [reply]
@islasimpson pls circle back when you have a closer look, since I'm about to derail the discussion for a moment. First, DQCORE is a custom variable and so needs a SourceMod (for now), CMFMCDZM is now CMFMC_DP. I suspect the V200, V850 errors are because there is no corresponding U200, U850, since they are computed using vector interpolations IIRC. None of this really matters; omitting them from this CAM-CHEM simulation is fine as they are primarily for our physics development runs.

@tilmes your case suggests you are running ne30pg2? I don't think this has ever been vetted for CAM-CHEM, has it? In this case reactions occur on the pg2 grid, and species are advected on the CSLAM, pg3 grid. I'm not necessarily opposed to pg2, Im just surprised to see you using it. Since it's not settled whether our workhorse is going to be pg2 or pg3, I don't know that forcing our future development runs with specified chemistry / radiative constituents from a pg2 run is something we should be committing to right now.

Also can we change the case names from "L48+BL10" to just "L58", for brevity and sanity, going forward?

**Reply by tilmes** [reply]
I setup the case the same as  what Cecile's run is, and it was a ne30pg2. I can look at the chemistry and see if it looks reasonable. If you prefer me setting up a pg3 run, I can do that as well.

**Reply by adamrher** [reply]
ah, scrolling up I see that @cecilehannay (accidentally?) pointed you to the pg2 run. Lets just run ol'fashion pg3 to be conservative. Please change the grid alias in create_newcase to be ```ne30pg3_ne30pg3_mg17```.

**Reply by tilmes** [reply]
Sure, that should be no problem! I could try running the pg2 for a couple years to see how it looks like ...

**Reply by dan800** [reply]
Simone, 5 days of computer time that needs to be redone is nothing to sneeze at - please do verify the UBC and update if necessary before running. The main reason to do this is to get the ozone heating correct in the UTLS and to have a water vapor baseline to compare no chemistry runs with. If this run is fundamentally flawed then it's not a good standard to tune to. 

BTW, I am confused by your statement that CAM-CHEM does not have an upper boundary condition - surely that is not right?

**Reply by tilmes** [reply]
@cecilehannay could you point me to the pg3 run, so I can make sure to set it up correctly?

**Reply by swrneale** [reply]
@cecilehannay do we have a pg3 versus simulation diagnostics comparison somewhere?

**Reply by JulioTBacmeister** [reply]
I agree with @adamrher.  For now let's stick to ne30pg3.

**Reply by islasimpson** [reply]
Here are some plots comparing what we get for T and H2O using WACCM (left), L83 with SC-WACCM and L83 with CAM.  Definitely seems like CAM could look nicer if we use a WACCM file to specify the upper boundary.

![L83_temperatures](https://user-images.githubusercontent.com/41842751/154410205-152d0ad8-cbca-44aa-ad1b-226d7e368519.png)
![L83_H2O](https://user-images.githubusercontent.com/41842751/154410214-ae589249-b050-4e3c-a1ab-2dc8be93fb3e.png)

**Comment by JulioTBacmeister** [comment]
As far as UBCs for water vapor and other species- we need to look at several different processes: 1) the vertical remapping/advection; 2) CLUBB's vertical diffusion; and 3) vertical diffusion in the GW scheme.   Anything else?

**Reply by dan800** [reply]
From @fvitt: My understanding is that all the transported tracers have zero fluxes across the upper boundary in "low-top" CAM and CAM-Chem.
I think we can do better and unify the L93 and L58 approach to boundary conditions by taking them from a prior WACCM run.

**Reply by dan800** [reply]
@islasimpson @tilmes and others, at the prompting of @fvitt, I have created an issue on ESCOMP/CAM
https://github.com/ESCOMP/CAM/issues/533#issue-1140373247
Please let me know if you see issues with adding such an enhancement.

**Comment by cecilehannay** [comment]
I am a bit confused. As far as I can tell the run #53 :
/glade/p/cesmdata/cseg/runs/cesm2_0/f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001
is a pg3 run. 
Am I missing something? @adamrher: could you check

**Reply by adamrher** [reply]
@cecilehannay if you scroll up you wrote:

> Option 1 is the run:
> https://github.com/NCAR/amwg_dev/issues/53
> 
> The case directory is at:
> 
>  /glade/p/cesmdata/cseg/runs/cesm2_0/f.e21.FWscHIST.ne30pg2_L48_BL10_cam6_3_041_control.hf.001

which is not run #53. @tilmes you can double check your settings w/ #53 proper:
/glade/p/cesmdata/cseg/runs/cesm2_0/f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001

Or take my word that all you need to do is change the grid alias in create_newcase. That is the only difference.

**Reply by cecilehannay** [reply]
Thanks. Yes, we should use: #53. Sorry for the mixed up. I edited the comment above.

**Comment by cecilehannay** [comment]
@swrneale: we have done a comparison pg2 (#58) versus pg3 (#53) at:
https://github.com/NCAR/amwg_dev/issues/58

**Comment by cecilehannay** [comment]
> Also can we change the case names from "L48+BL10" to just "L58", for brevity and sanity, going forward?

@adam: 
- yes, we had decided to do that based on @dan800's request. 
- I used L48_BL10 for the last run done with cam3_06_041 for consistency in naming convention with the same tag. 
- But I am planning to update the naming convention when we move to the next tag (i.e to use L58 and to drop hf as all our runs are hf right now).

**Comment by tilmes** [comment]
Here is the updated case: /glade/p/cesm/chwg_dev/tilmes/cases/cesm2_ch/CAM7_tests/f.e21.FWscHIST_chemistry.ne30pg3_L58_cam6_3_041_control.001
It is currently running for a month.
Please let me know if we want any more changes at this point, or if I should start the 20 years?

**Reply by tilmes** [reply]
@dan800 I know we need to add the UBC and then we can do a comparison run.

**Reply by dan800** [reply]
OK. Francis is working on coding that up so it won't be available this week and I think it good to get a L58 ozone into CAM L58. Until we have an UBC option we will have to go with the standard zero flux boundary conditions for CAM-CHEM runs. Can you please rename this something less confusing than FWscHIST_chemistry - it's not sc or W! This should really be FCHISTL58 or similar (i.e., a low top historical run with chemistry). Thanks.

**Reply by tilmes** [reply]
Thanks! Model is running, sorry I kept the name for now, so we remember where this setup is coming  from (adding chemistry to the WACCM-SC run), but I agree to change it next.

**Comment by tilmes** [comment]
@islasimpson I have checked the GW setting in the L58 model and you are right, the settings are the same, just we lat the momentum get through the top in the low top model. I had at one point worked on this with Anne Smith to improve this for CAMchem. I will follow-up on this.

**Reply by tilmes** [reply]
@islasimpson @JulioTBacmeister    I cam comparing a standard CAM vs CAMchem compset and these are the differences  in the GW we have currently (everything  else is the same), maybe  you could have a look and see if we need to change something to the L58model or if those differences don't matter?
CAM (32L  FV)
gw_apply_tndmax                = .true. 
gw_dc          = 0.D0 
gw_limit_tau_without_eff               = .false.
gw_lndscl_sgh          = .true. 
gw_oro_south_fac               = 1.d0 
pgwv           = 0

CAMchem (32L FV):
effgw_beres_dp         = 0.5D0 
effgw_cm               = 1.D0 
frontgfc               = 3.0D-15 
gw_apply_tndmax                = .false.
gw_dc          = 2.5D0 
gw_drag_file           = '/glade/p/cesmdata/cseg/inputdata/atm/waccm/gw/newmfspectra40_dc25.nc'
gw_limit_tau_without_eff               = .true.
gw_lndscl_sgh          = .false. 
gw_oro_south_fac               = 2.d0 
pgwv           = 32

**Reply by islasimpson** [reply]
Do you mean that we'd want to change these things in ongoing tests so that there is some consistency with what was done in the chemistry simulation?  I don't have a good sense of what most of these parameters do, so @JulioTBacmeister is probably better placed to answer about this.  But I could see that if we were to start all tests anew with chemistry prescribed in this way then it would make sense to be consistent from now on unless there's a reason that the parameters that are currently used in CAM are preferable for the climate.

**Reply by JulioTBacmeister** [reply]
The parameters used in the CAMchem namelist will only have an effect if the frontal and convective GW sources are being used, but I don't see the settings 

 use_gw_convect_dp		= .true.
 use_gw_front		= .true.

in the CAM chem namelist which should be there if these GW sources are intended to be used.   Now that we are on the topic I'll note that the gw_limit_tau_without_eff logical is not used anywhere so we should get rid of it. The parameters  gw_lndscl_sgh  and gw_oro_south_fac are only used by the previous orographic GW scheme not the current one.  Clean up is in order.

If we want CAMchem to use nonorographic GW then we should compare with a WACCM namelist.

**Comment by tilmes** [comment]
@cecilehannay This run has now 20 years completed: /glade/u/home/tilmes/archive/ f.e21.FWscHIST_chemistry.ne30pg3_L58_cam6_3_041_control.001
Do we want to compare  to the run without chemistry? Would you be able  to run the AMWG  diagnostics? 
After  that  I can look into producing  the forcing files for CAM.

**Reply by cecilehannay** [reply]
@tilmes : I am running the diags. 

Do you want to me to add this case as an "issue" so it will be part of our suite of runs on github?

**Reply by cecilehannay** [reply]
@tilmes 

Diags are: https://webext.cgd.ucar.edu/FWscHIST/f.e21.FWscHIST_chemistry.ne30pg3_L58_cam6_3_041_control.001/atm/f.e21.FWscHIST_chemistry.ne30pg3_L58_cam6_3_041_control.001.1980_1989-f.e21.FWscHIST.ne30_L48_BL10_cam6_3_041_control.hf.001.1980_1989/

**Reply by tilmes** [reply]
@cecilehannay Thanks, here are also the diagnostics for comparisons with observations: /glade/scratch/tilmes/archive/f.e21.FWscHIST_chemistry.ne30pg3_L58_cam6_3_041_control.001/atm/proc/diag/f.e21.FWscHIST_chemistry.ne30pg3_L58_cam6_3_041_control.001-obs.1980_1999
You  can add this as an "issue" to the suite of runs on github, it may be useful for later comparisions.

**Comment by tilmes** [comment]
@cecilehannay we are running some test cases for improving the stratospheric water vapor, and using a nudged 58layer and 32layer model run. Running nudged means that we are turning the GREGORIAN calendar on. This triggers an error in the land that has been addressed here: https://github.com/ESCOMP/CTSM/issues/1657
I wonder if you could update your clm version in Externals.cfg to ctsm5.1.dev073, to resolve this problem?  
Thanks Simone

**Reply by adamrher** [reply]
@cacraigucar we have a request to update our land externals. But they are running cam6_3_041, and we don't want them to merge to the head because the PUMAS commit cam6_3_046 is still being evaluated for potential problems (and we made it hard to turn off pumas w/ cam_dev in cam6_3_046). Do you think they can just swap in new ctsm externals to their cam6_3_041 sandbox, instead?

**Reply by cacraigucar** [reply]
> @cacraigucar we have a request to update our land externals. But they are running cam6_3_041, and we don't want them to merge to the head because the PUMAS commit cam6_3_046 is still being evaluated for potential problems (and we made it hard to turn off pumas w/ cam_dev in cam6_3_046). Do you think they can just swap in new ctsm externals to their cam6_3_041 sandbox, instead?

In theory this should work.  Looking back at the tags, I believe that any tag before ctsm5.1.dev075 should work with cam6_3_041.  Fire emission updates were put into ctsm5.1.dev075 and cam6_3_042 and I believe moving to these tags or later would need to be updated in tandem (along with other externals as well).  But since you are not proposing moving that far, it appears that it should work.  But since this configuration was never tested, I can't say with 100% certainty that it will be okay, just that I've not seen an issue.

**Reply by cecilehannay** [reply]
I will try to switch to the land tag and see if I run into any issue.

**Reply by tilmes** [reply]
Thanks, let me know once you have done it!

**Reply by adamrher** [reply]
It seems to work; ran 1 day of cam6_3_041 w/ the ctsm5.1.dev073 external swapped in. @cecilehannay I don't think we should modify your original cam6_3_041 sandbox. Should @tilmes just use my sandbox: /glade/u/home/aherring/src/cam6_3_041.newctsm ? Or do you want the code in your directory?

**Comment by cecilehannay** [comment]
Thanks.  

Quick question: do we want to use that code for future runs ? 

Also: for reference, I don’t modify the original sandbox when we have done any run for that sandbox.  
I create a new sandbox instead. This is to ensure there is always a correspondance between the sandbox and the case directory.

**Reply by tilmes** [reply]
@cecilehannay Thanks, could you point me to that sandbox?

**Reply by adamrher** [reply]
> Quick question: do we want to use that code for future runs ?

I don't have strong feelings either way ... Maybe we consider this a one off for Simone's nudged runs? My hope is that we're all eventually going to be running the head of cam_development, which would ideally have these new land externals.
